airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

```diff
@@ -0,0 +1,188 @@
+
+
+class DestinationNameTransformer:
+    """
+    Handles naming conventions in destinations for all kind of sql identifiers:
+    - schema
+    - table
+    - column
+    """
+
+    def __init__(self, integration_type: DestinationType):
+        """
+        @param integration_type is the destination type of warehouse
+        """
+        self.integration_type: DestinationType = integration_type
+
+    # Public methods
+
+    def needs_quotes(self, input_name: str) -> bool:
+        """
+        @param input_name to test if it needs to manipulated with quotes or not
+        """
+        if is_reserved_keyword(input_name, self.integration_type):
+            return True
+        if self.integration_type.value == DestinationType.BIGQUERY.value:
+            return False
+        doesnt_start_with_alphaunderscore = match("[^A-Za-z_]", input_name[0]) is not None
+        contains_non_alphanumeric = match(".*[^A-Za-z0-9_].*", input_name) is not None
+        return doesnt_start_with_alphaunderscore or contains_non_alphanumeric
+
+    def normalize_schema_name(self, schema_name: str, in_jinja: bool = False, truncate: bool = True) -> str:
+        """
+        @param schema_name is the schema to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        @param truncate force ignoring truncate operation on resulting normalized name. For example, if we don't
+        control how the name would be normalized
+        """
+        return self.__normalize_non_column_identifier_name(input_name=schema_name, in_jinja=in_jinja, truncate=truncate)
+
+    def normalize_table_name(self, table_name: str, in_jinja: bool = False, truncate: bool = True) -> str:
+        """
+        @param table_name is the table to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        @param truncate force ignoring truncate operation on resulting normalized name. For example, if we don't
+        control how the name would be normalized
+        """
+        return self.__normalize_non_column_identifier_name(input_name=table_name, in_jinja=in_jinja, truncate=truncate)
+
+    def normalize_column_name(self, column_name: str, in_jinja: bool = False, truncate: bool = True) -> str:
+        """
+        @param column_name is the column to normalize
+        @param in_jinja is a boolean to specify if the returned normalized will be used inside a jinja macro or not
+        @param truncate force ignoring truncate operation on resulting normalized name. For example, if we don't
+        control how the name would be normalized
+        """
+        return self.__normalize_identifier_name(column_name=column_name, in_jinja=in_jinja, truncate=truncate)
+
+    # Private methods
+
+    def __normalize_non_column_identifier_name(self, input_name: str, in_jinja: bool = False, truncate: bool = True) -> str:
```

Comment:
You're not using in_jinja here. expected?

Reply:
it's not used yet in the code no but it's to be consistent in the public API...

i guess it needs a little more cleaning work there